models/resnet.py

```diff
@@ -99,9 +128,8 @@ def load_from_pretrained(self):
       assert obj.shape == dat.shape, (k, obj.shape, dat.shape)
       obj.assign(dat)
 
-ResNet18 = lambda num_classes=1000: ResNet(BasicBlock, [2,2,2,2], num_classes=num_classes, url='https://download.pytorch.org/models/resnet18-5c106cde.pth')
-ResNet34 = lambda num_classes=1000: ResNet(BasicBlock, [3,4,6,3], num_classes=num_classes, url='https://download.pytorch.org/models/resnet34-333f7ec4.pth')
-ResNet50 = lambda num_classes=1000: ResNet(Bottleneck, [3,4,6,3], num_classes=num_classes, url='https://download.pytorch.org/models/resnet50-19c8e357.pth')
-ResNet101 = lambda num_classes=1000: ResNet(Bottleneck, [3,4,23,3], num_classes=num_classes, url='https://download.pytorch.org/models/resnet101-5d3b4d8f.pth')
-ResNet101 = lambda num_classes=1000: ResNet(Bottleneck, [3,8,36,3], num_classes=num_classes, url='https://download.pytorch.org/models/resnet152-b121ed2d.pth')
-
+ResNet18 = lambda num_classes=1000: ResNet(18, num_classes=num_classes)
+ResNet34 = lambda num_classes=1000: ResNet(34, num_classes=num_classes)
+ResNet50 = lambda num_classes=1000: ResNet(50, num_classes=num_classes)
+ResNet101 = lambda num_classes=1000: ResNet(101, num_classes=num_classes)
```

Comment:
btw trying to use resnet bigger than 50 causes RecursionError in the tensor.py deepwalk. This occurs whether or not you load pretrained weights. This is some issue bigger than this refactor.